Adds HTTP Data Structure

imdominicreed opened the pull request:
We need data structure to organize the HTTP data. This adds it for POST
and GET since these are the main two that we will currently be using.

Commit:
Adds HTTP Data Structure

We need data structure to organize the HTTP data. This adds it for POST
and GET since these are the main two that we will currently be using.

Commit:
Adds HTTP Data Structure

We need data structure to organize the HTTP data. This adds it for POST
and GET since these are the main two that we will currently be using.

evanugarte commented on http.hh:
```diff
@@ -0,0 +1,19 @@
+
+struct http_post : http_request {
+  std::string url;
+  int fd;
```
what is fd? can we give it a better name?

imdominicreed replied:
its the file descriptor where we will store the binary data.

imdominicreed replied:
ill change it to file_descriptor

evanugarte replied:
wouldn't that be generated on the server side? we might not need this here and can just use the `http_request` struct for both get and post

imdominicreed replied:
i was thinking that the serverside parses the requests into these data structures and passes them to the handlers.

imdominicreed replied:
then we can have the lambdas that take in the http request as a parameter and returns some http response.

evanugarte commented on http.hh:
```diff
@@ -0,0 +1,19 @@
+};
+
+struct http_get : http_request {
+  std::string url;
```
`url` is in both structs lets move it to `http_request`

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
Adds HTTP Data Structure

We need data structure to organize the HTTP data. This adds it for POST and GET since these are the main two that we will currently be using.

evanugarte commented on http.h:
```diff
@@ -0,0 +1,18 @@
+#include <string>
+#include <unordered_map>
+
```
this should probably live inside of a namespace if were gonna follow the google style guide (see this section on [namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces))

we could do something like

```cpp
namespace domino {
namespace http {
  enum Method { kGet, kPost };
  struct request {
    Method method;
    // if the url was http://localhost:3000/upload?param=1
    // the endpoint should just be "upload"
    std::string endpoint;
  };
} // namespace http
} // namespace domino
```

then things can get super clean like

```cpp
domino::http::method::kGet;
domino::http::method::kPost;

domino::http::request request;

request.method
request.endpoint;
```

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
erge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
erge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds

Commit:
Merge branch 'http_ds' of https://github.com/imdominicreed/just-a-random-webserver into http_ds
